src/qibosoq/qick_programs.py

```diff
@@ -0,0 +1,364 @@
+""" QickPrograms used by qibosoq to execute sequences and sweeps """
+
+from abc import ABC, abstractmethod
+from dataclasses import asdict
+from typing import List, Tuple
+
+import numpy as np
+from qibolab.instruments.rfsoc import QickProgramConfig
+from qibolab.platforms.abstract import Qubit
+from qibolab.pulses import Drag, Gaussian, Pulse, PulseSequence, PulseType, Rectangular
+from qibolab.sweeper import Parameter, Sweeper
+from qick import AveragerProgram, QickSoc, RAveragerProgram
+
+# conversion coefficients (in qibolab we use Hz and ns)
+HZ_TO_MHZ = 1e-6
+NS_TO_US = 1e-3
+
+
+class GeneralQickProgram(ABC):
+    """Abstract class for QickPrograms"""
```

Comment:
I am wondering if this class should inherit from some `qick` abstract class, since you are basically overriding/defining methods that qick requires and in some cases (`self.di_buf` or `self.us2cycles`) you are using qick methods in this class. 

I understand that you cannot inherit the `AveragerProgram` because you may need the `RAveragerProgram` instead, but maybe the base class of these. This may complicate the inheritance structure though and I am not sure how well the double inheritance below will work.

Reply:
Looking at pylint, this does not work as intended... The super.acquire function (called in the acquire function of GeneralQickProgram) should call the acquire either of AveragerProgram or of RAveragerProgram, but if the GeneralQickProgram inherits from QickProgram then the QickProgram.acquire function gets called...

If I don't add the QickProgram, however, it is true that pylint complains for self.di_buf, self.us2cycles and many more